modules/core/errors/errors.go

```diff
@@ -60,4 +60,7 @@ var (
 
 	// ErrNotFound defines an error when requested entity doesn't exist in the state.
 	ErrNotFound = errorsmod.Register(codespace, 16, "not found")
+
+	// ErrInvalidAuthority defines an error when the authority address is invalid.
+	ErrInvalidAuthority = errorsmod.Register(codespace, 17, "invalid authority")
```

Comment:
mega nit tbh, but I don't see why we couldn't use `ErrUnauthorized` 

happy to keep it for parity with sdk